src/inverted_index.c

```diff
@@ -443,7 +517,8 @@ size_t InvertedIndex_WriteEntryGeneric(InvertedIndex *idx, IndexEncoder encoder,
           INDEX_BLOCK_SIZE_DOCID_ONLY;
 
   // see if we need to grow the current block
-  if (blk->numDocs >= blockSize) {
+  if (blk->numDocs >= blockSize && !same_doc) {
```

Comment:
IMO, this should be by numEntries.
If docs have multiple values, we don't want to scan many to reach our doc.

Reply:
This was already discussed, and it is also mentioned in the description of the PR:
> We add a new block when a number of docs is reached - not values 
...
If we change that, and the same doc can span more than a single block - need to adjust IndexReader_SkipToBlock (and maybe more)

I suggest to keep the current approach for now and we can review that later on.
(BTW, currently `IndexBlock` has no `numEntries`, only `numDocs`)

Regarding scanning, if we have something like the following in an inverted index with some range:
In Block1 we have doc1 val1 and doc1 val2 and doc1 val3 and doc1 val4
In Block2 we have doc2  val1 and doc2 val3
Or
doc1 is spanning 2 blocks, in something like this:
In Block1 we have doc1 val1 and doc1 val2
In Block2 we have doc1 val3 and doc1 val4
In Block3 we have doc2  val1 and doc2 val3

Won't we still need to scan the same to reach values from doc2?

Reply:
After further discussion, the comment is not about splitting entries from the same doc into multiple blocks, rather splitting to new block when entries count exceeds blockSize and entries are not from the same doc.
Good suggestion! (requires adding to `IndexBlock` a new member `numEntries`)